gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/util/DefaultTraversal.java

```diff
@@ -350,6 +351,10 @@ public void setGraph(final Graph graph) {
         this.graph = graph;
     }
 
+    public void setLocked(boolean locked) {
```

Comment:
what is the intent of this change? i don't know what folks might do with that, but "unlocking" a traversal might open up bad/unexpected things for users....

Reply:
It's a separate conversation. I will soon raise a PR for that. I think that the way in which we apply strategies (first to all steps and then to each children one at a time), could be improved to apply all strategies at the same time to the entire traversal and sub-traversal. I will remove this change from this PR, pending further discussion on a new PR.

Reply:
Resolving this comment since I removed the file being discussed from the scope of this PR.

Reply:
note that strategy application changed/improved on `master` for 3.5.0 already:

https://tinkerpop.apache.org/docs/3.5.0-SNAPSHOT/upgrade/#_traversalstrategy_application

looking forward to hearing what you think we might change but we will need to be careful not to introduce a change that inadvertently changes traversal behavior because we change strategy application order/approach. that's why we saved the above referenced change for 3.5.0.